pkgs/development/libraries/highfive/default.nix

```
@@ -0,0 +1,50 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, eigen
+, hdf5
+}:
+
+let
+  version = "2.2";
+  mpiSupport = hdf5.mpiSupport;
+  mpi = hdf5.mpi;
+in
+stdenv.mkDerivation {
+  pname = "highfive";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "BlueBrain";
+    repo = "HighFive";
+    rev = "4c70d818ed18231563fe49ff197d1c41054be592";
+    sha256 = "02xy3c2ix3nw8109aw75ixj651knzc5rjqwqrxximm4hzwx09frk";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  propagatedBuildInputs = [ boost eigen hdf5 ];
```

unless these need to be propagated to child derivations, I think buildInputs is what you want
```suggestion
  buildInputs = [ boost eigen hdf5 ];
```